content/docs/tasks/telemetry/kiali/index.md

```diff
@@ -0,0 +1,400 @@
+---
+title: Visualizing Your Mesh
+description: This task shows you how to visualize your services within an Istio mesh.
+weight: 49
+keywords: [telemetry,visualization]
+---
+
+This task shows you how to visualize different aspects of your Istio mesh.
+
+As part of this task, you install the [Kiali](https://www.kiali.io) add-on
+and use the web-based graphical user interface to view service graphs of
+the mesh and your Istio configuration objects. Lastly, you use the Kiali
+Public API to generate graph data in the form of consumable JSON.
+
+The [Bookinfo](/docs/examples/bookinfo/) sample application is used as the
+example application throughout this task.
+
+## Before you begin
+
+Create a secret in your Istio namespace with credentials that you use to
+authenticate to Kiali. See the
+[Helm README](https://github.com/istio/istio/blob/master/install/kubernetes/helm/istio/README.md#installing-the-chart)
+for details, but here is an example command you can modify and run to create a secret:
+
+```bash
```

Comment:
I originally did have it this way, however, there is a bug in the text_bash markup implementation that caused this to not render properly. I mentioned this in this comment: https://github.com/istio/istio.io/pull/2715#discussion_r222068150

The problem is text bash markup does not render a "here document" properly. Using "``bash" does render it properly; doing it this way allows the reader to still easily click the "copy" button and get all the commands on the clipboard so it can be pasted to a console with a simple "paste" operation as opposed to copy-n-pasting each line individually (while not selecting the $ at the same time) - that would be what the reader would need to do if I put "$" prompt in front of the individual commands inside "```bash".

For comparison, here's the bug - notice when using text_bash, the "here document" is rendered as if it is the command output, so it looks like it is output and not part of the command and the "copy" button doesn't copy  the "here document" into the clipboard:

![image](https://user-images.githubusercontent.com/2029470/46629632-90421d00-cb0f-11e8-8ba2-740b4a69fa2a.png)


 Thus, when using "here documents", it looks nicer with "```bash":

![image](https://user-images.githubusercontent.com/2029470/46629655-a6e87400-cb0f-11e8-8bfa-bde752d7ed8a.png)

Reply:
I just tried adding a "$" to the beginning of the commands in this "```bash" block, but the copy-n-paste doesn't work. If a reader clicks the "copy" button and tries to paste it into a console to create the secret, a bunch of errors will result:

```
$ $ USERNAME=$(echo -n 'admin' | base64)
bash: $: command not found...
$ $ PASSPHRASE=$(echo -n 'mysecret' | base64)
bash: $: command not found...
$ $ NAMESPACE=istio-system
bash: $: command not found...
$ $ kubectl create namespace $NAMESPACE
bash: $: command not found...
$ $ cat <<EOF | kubectl apply -f -
...
> EOF
bash: $: command not found...
error: no objects passed to apply
```

So there are problems all around - can't use text_bash because of a bug in rendering "here documents", and can't put "$" in the "```bash" section without making the copy-paste feature fail for the user. To make it easier for the reader to run this task, the way I have it now seems the best. The reader need only press the "copy" button and "paste" the bash content into their bash command window and the secret will be created.

Reply:
The copy button functionality is the only thing I would considered bugged. Having the configuration file as part of the command is considered a walk-around ideally you would have the command be something like:
```bash
kubectl apply -f this-conf.yaml
```
instead.